### DISCUSSION TITLE
Conditional controls for relays in Nixie

### DISCUSSION THREAD

**Discussion by Yankus** [tagyoureit/nodejs-poolController#919] [Q&A] [answered]
Starting to setup my Nixie stand alone system to control my pool. I have a raspberry pi 5, Eight Relays 4A/120V 8-Layer Stackable HAT, and Building Automation V4 8-Layer Stackable HAT currently on the bench for testing. So far everything is working by following the WiKi "DIY Standalone Nixie Pool Controller". Trying to stick to be basics now, and get the pool pump running. I have a Pentair SuperFlo VS, part 342001 made on 8/8/2019. Almost made it.. Darn. From what I understand the cable puts out 5 volts, and I have 4 wires to trigger the 'speeds'. In Nixie I have the pump configured as "SuperFlo VS" with the relays and circuits configured for the different speeds. All 4 relays operate independently from each other. 

Is there a way to have a condition in relay manager, pool controller or whatever to say if relay 1 is on, 2,3 and 4 must be off? As I have it setup now all relays can be on at once. I'm not sure how that will affect my pool pump (nor do I want to find out!). Basically I want to force it to only run one speed at a time.

**Comment by rstrouse** [comment, marked as answer]
If you have configured the relays as SuperFlo then njsPC will take care of the relay states.  That being said the SuperFlo has no such requirement and will run at the highest detectable speed based on the relay settings.  The relay sequencing is also managed via njsPC so that it does not shut off between speeds and therefore trigger a prime.

**Comment by Yankus** [comment]
Thanks for the reply. So my worries about having multiple speeds triggered to the pump doesn't matter. It will take the top speed and apply that. I was going to play with circuit groups, have speed 1 triggered but as part of the group have speed 2,3, and 4 turn off. I wonder if there is a delay between the relays and not trigger a prime in the pump. Good point on that too.
